azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java

```diff
@@ -5,34 +5,176 @@
+
+                for (SubscriptionEntity s : subscriptionsOnTenant) {
+                    String key = StringUtils.lowerCase(s.getId());
+                    if (subscriptionMap.putIfAbsent(key, s) != null) {
+                        validTenantIds.add(tenantId);
+                    }
+                }
+            } catch (Exception ex) {
+                // ignore AuthenticationException since on some tenants, it doesn't allow list subscriptions
+                if (!(ExceptionUtils.getRootCause(ex) instanceof AuthenticationException)) {
+                    this.entity.setAuthenticated(false);
+                    Account.this.entity.setError(ex);
+                }
+
+            }
+        });
+
+        // since some tenants doesn't have subscriptions, reduce the tenants
+        this.entity.setTenantIds(new ArrayList<>(validTenantIds));
+        this.entity.setSubscriptions(new ArrayList<>(subscriptionMap.values()));
+        if (subscriptionMap.isEmpty()) {
+            this.entity.setAuthenticated(false);
+        }
+    }
+
+    public void selectSubscriptions(List<String> subscriptionIds) {
+        // select subscriptions
+        if (CollectionUtils.isNotEmpty(subscriptionIds) && CollectionUtils.isNotEmpty(this.entity.getSubscriptions())) {
+            entity.getSubscriptions().stream().filter(s ->
+                    Utils.containsIgnoreCase(subscriptionIds, s.getId())
+            ).forEach(t -> {
+                t.setSelected(true);
+            });
+        }
+        entity.setSelectedSubscriptions(entity.getSubscriptions().stream().filter(SubscriptionEntity::isSelected).collect(Collectors.toList()));
+    }
+
+    private SubscriptionEntity toSubscriptionEntity(AzureEnvironment env, String tenantId, Subscription subscription) {
+        final SubscriptionEntity subscriptionEntity = new SubscriptionEntity();
+        subscriptionEntity.setId(subscription.subscriptionId());
+        subscriptionEntity.setName(subscription.displayName());
+        subscriptionEntity.setTenantId(tenantId);
+        subscriptionEntity.setEnvironment(env);
+        return subscriptionEntity;
+    }
+
+    public boolean isAuthenticated() {
+        return this.entity != null && this.entity.isAuthenticated();
+    }
+
+    public List<SubscriptionEntity> getSubscriptions() {
+        if (this.entity != null) {
+            return this.entity.getSubscriptions();
+        }
+        return null;
+    }
+
+    public List<SubscriptionEntity> getSelectedSubscriptions() {
+        if (this.entity != null) {
+            return this.entity.getSelectedSubscriptions();
+        }
+        return null;
+    }
+
+    public TokenCredential getCredential(String subscriptionId) throws LoginFailureException {
+        if (!this.isAuthenticated()) {
+            throw new LoginFailureException("Please login first.");
+        }
+        Objects.requireNonNull(this.credentialBuilder, "Azure Account should be initialized first.");
```

Comment:
Change messages to login first? As seems we will not do nothing for credentialBuilder in `initialize()`

Reply:
It is fine to have null credentialBuilder when login fails